lib/common/ipc_client.c

```diff
@@ -830,8 +830,7 @@ crm_ipc_connect(crm_ipc_t * client)
         errno = ECONNABORTED;
         return FALSE;
 
-    } else if (rv < 0) {
-        errno = -rv;
+    } else if (rv != pcmk_rc_ok) {
         crm_perror(LOG_ERR, "Could not verify authenticity of daemon (IPC %s)",
                    client->name);
         crm_ipc_close(client);
```

Comment:
errno is being set to -rv just after this, which will no longer be meaningful ... we can set it to rv if rv > 0 (positive values are errnos) and ENOTCONN otherwise

Reply:
/me looks to see if there are any more worms in this can